pkg/plugins/manager.go

```diff
@@ -11,160 +12,104 @@ import (
 )
 
 var (
-	errPluginGroupNotFound = fmt.Errorf("unknown plugin group")
-	errPluginTypeNotFound  = fmt.Errorf("unknown plugin type")
+	ErrInvalidPluginNotFound = errors.New("invalid plugin type")
+	ErrPluginNameNotFound    = errors.New("unknown plugin name")
+	ErrUninitializedManager  = errors.New("uninitialized plugins manager")
 )
 
-// PluginType specifies the type of related plugins
-// For example, "diff" is a type of plugins that know how to perform a diff.
-type PluginType string
-
-// PluginName specifies the name of a plugin
-// For example, "delta" is a plugin.
-type PluginName string
+var allowedProtocols = []plugin.Protocol{
+	plugin.ProtocolGRPC,
+}
 
-// PluginIdentity identifies the plugin implementation.
+// PluginIdentity identifies the plugin's version and executable location.
 type PluginIdentity struct {
-	Impl               plugin.Plugin
-	ImplName           string
-	HandshakeConfig    plugin.HandshakeConfig
+	Version            int
 	ExecutableLocation string
 }
-type pluginTypeConfigMap map[PluginName]plugin.ClientConfig
 
-// PluginTypeNameMap example:
-/*
-{
-	"diff": {
-		"delta": plugin.clientConfig{
-			HandshakeConfig: plugin.HandshakeConfig{
-				ProtocolVersion:  1,
-				MagicCookieKey:   "key",
-				MagicCookieValue: "value",
-			},
-			Cmd: exec.Command("path/to/executable/diff/delta"),
-			AllowedProtocols: []plugin.Protocol{
-				plugin.ProtocolGRPC,
-			},
-			Impl: map[string]plugin.Plugin{
-				"deltaDiff": DeltaGRPCPlugin,
-			},
-		},
-		"iceberg: {
-			HandshakeConfig: plugin.HandshakeConfig{
-				ProtocolVersion:  1,
-				MagicCookieKey:   "key",
-				MagicCookieValue: "value",
-			},
-			Cmd: exec.Command("path/to/executable/diff/iceberg"),
-			AllowedProtocols: []plugin.Protocol{
-				plugin.ProtocolGRPC,
-			},
-			Impl: map[string]plugin.Plugin{
-				"icebergDiff": IcebergGRPCPlugin
-			},
-		}
-	},
-	"merge": {
-		"delta": {
-			HandshakeConfig: plugin.HandshakeConfig{
-				ProtocolVersion:  1,
-				MagicCookieKey:   "key",
-				MagicCookieValue: "value",
-			},
-			Cmd: exec.Command("path/to/executable/merge/delta"),
-			AllowedProtocols: []plugin.Protocol{
-				plugin.ProtocolGRPC,
-			},
-			Impl: map[string]plugin.Plugin{
-				"deltaMerge": DeltaGRPCPlugin,
-			},
-		},
-		"iceberg: {
-			HandshakeConfig: plugin.HandshakeConfig{
-				ProtocolVersion:  1,
-				MagicCookieKey:   "key",
-				MagicCookieValue: "value",
-			},
-			Cmd: exec.Command("path/to/executable/merge/iceberg"),
-			AllowedProtocols: []plugin.Protocol{
-				plugin.ProtocolGRPC,
-			},
-			Impl: map[string]plugin.Plugin{
-				"icebergMerge": IcebergGRPCPlugin
-			},
-		}
-	},
+// PluginAuth includes authentication properties for the plugin.
```

Comment:
Can you expand?  It looks like a _single_ property, and it's not clear what the property means.  Maybe if the docs added what "authentication" we're performing it would be easier for me to understand.

Reply:
It's a go-plugin thing for insuring that the client and server of the plugin identify one another